src/data-structures/linked-list/LinkedList.js

```diff
@@ -207,4 +207,88 @@ export default class LinkedList {
   toString(callback) {
     return this.toArray().map(node => node.toString(callback)).toString();
   }
+
+  /**
+   * Traverse through all nodes of the list from head to tail
+   * @param {*} callback
+   * @return {LinkedListNode[]}
+   */
+  traverse(callback = undefined) {
+    const traversedNodes = [];
+    let currentNode = this.head;
+
+    while (currentNode) {
+      // If callback is specified then try to call the callback.
+      if (callback) {
+        callback(currentNode);
+      } else {
+        traversedNodes.push(currentNode);
+      }
+      currentNode = currentNode.next;
+    }
+
+    return traversedNodes;
+  }
+
+  /**
+   * The items in the list have been traversed in reverse order
+   */
+  reverseTraversal(callback = undefined) {
```

Comment:
This implementation of `reverseTraversal` has a running time of `O(n^2)`. It would be more straightforward and more performant (`O(n)`) to just define `reverseTraversal` as `traverse().reverse()`.

If you want to write a `reverseTraverse` function as an algorithm exercise, it might be better to define it recursively, like this:
```js
function reverseTraverse(node, callback) {
  if (typeof callback !== 'function') {
    throw new TypeError(
      `reverseTraverse method requires a callback function as an argument.\nArgument given: ${typeof callback}`
    )
  }

  if (!node) return [];

  return  reverseTraverse(
    node.next, callback
  ).concat(callback(node.value));
}
```